packages/dcos-launch/extra/launch/config.py

```diff
@@ -307,11 +311,14 @@ def calculate_instance_count(num_masters, num_private_agents, num_public_agents)
     return str(1 + int(num_masters) + int(num_private_agents) + int(num_public_agents))
 
 
-def validate_os_name(os_name, platform):
+def validate_os_name(os_name, platform, provider):
     # TODO DCOS-14033: [gen.internals] Source validate functions are global only
-    if platform == 'bare_cluster':
+    if provider != 'onprem':
+        # the os_name parameter only applies to homogenous clusters that need
+        # to be created by dcos-launch for an onprem install. In other cases,
+        # the deployment template is parameterized for OS selection
         return
-    elif platform == 'aws':
+    if platform == 'aws':
```

Comment:
Could this `if` reuse the `validate_onprem_provider_platform` method? If the `platforms` for `onprem` grew, this place would also need updating.